app/client/components/updatableAmount.tsx

```diff
@@ -0,0 +1,287 @@
+      subscription.plan.currency
+    }${currentAmount.toFixed(2)} a ${
+      subscription.plan.interval
+    }. Please enter a different amount.`;
+  }
+  const minAmount = calculateMinOrMaxAmount(
+    MinOrMax.minimum,
+    subscription.plan
+  );
+  if (newValue < minAmount) {
+    return `Please enter an amount of ${
+      subscription.plan.currency
+    }${minAmount} or more a ${subscription.plan.interval}`;
+  }
+  const maxAmount = calculateMinOrMaxAmount(
+    MinOrMax.maximum,
+    subscription.plan
+  );
+  if (newValue > maxAmount) {
+    return `Thank you but we cannot accept contributions over ${
+      subscription.plan.currency
+    }${maxAmount} a ${subscription.plan.interval}`;
+  }
+  if (newValue > 10 * currentAmount) {
+    return `Your current contribution is ${
+      subscription.plan.currency
+    }${currentAmount.toFixed(2)} a ${
+      subscription.plan.interval
+    }. ${pleaseCheck}`;
+  }
+};
+
+const amountValidationWarningCSS = {
+  ...validationWarningCSS,
+  minHeight: "2rem",
+  padding: "5px 15px",
+  width: "100%"
+};
+
+class UpdateAmountLoader extends AsyncLoader<string> {}
+
+const getAmountUpdater = (
+  newAmount: number,
+  productType: ProductType
+) => async () =>
+  await fetch(`/api/update/amount/${productType.urlPart}`, {
+    credentials: "include",
+    method: "POST",
+    mode: "same-origin",
+    body: JSON.stringify({ newPaymentAmount: newAmount })
+  });
+
+export type UpdatableAmountProps = WithSubscription & WithProductType;
+
+export interface UpdatableAmountState {
+  inEditMode: boolean;
+  isApplyingUpdate: boolean;
+  currentAmount: number;
+  newAmount: number;
+  validationMessage?: string;
+}
+
+// tslint:disable-next-line:max-classes-per-file
+export class UpdatableAmount extends React.Component<
+  UpdatableAmountProps,
+  UpdatableAmountState
+> {
+  public initialAmount = this.props.subscription.nextPaymentPrice / 100.0;
+  public state = {
+    inEditMode: false,
+    isApplyingUpdate: false,
+    currentAmount: this.initialAmount,
+    newAmount: this.initialAmount,
+    validationMessage: ""
+  };
+
+  public render(): React.ReactNode {
+    return (
+      <div css={wrappingContainerCSS}>
+        {this.state.isApplyingUpdate ? (
+          <UpdateAmountLoader
+            fetch={getAmountUpdater(
+              this.state.newAmount,
+              this.props.productType
+            )}
+            readerOnOK={(resp: Response) => resp.text()}
+            render={() => {
+              this.setState({
+                isApplyingUpdate: false,
+                inEditMode: false,
+                currentAmount: this.state.newAmount
+              });
+              return null;
+            }}
+            loadingMessage={"Updating..."}
+            errorRender={() => {
+              this.setState({
+                isApplyingUpdate: false,
+                validationMessage:
+                  "Updating failed this time. Please try again later..."
```

Comment:
We should add a card to analyse why this update might fail and whether this message is always going to be appropriate (definitely one for a follow up PR though)